MAUI/DataGrid/column-drag-and-drop.md

```diff
@@ -0,0 +1,226 @@
+
+The `QueryColumnDragging` event provides the following properties in the `DataGridQueryColumnDraggingEventArgs`:
+
+- `From`: Returns the index of the currently dragging column.
+- `To`: Returns the index where you are trying to drop the column.
+- `DraggingAction`: Returns the column dragging details as a `DataGridDragAction`.
+- `DraggingPosition`: Returns the position of the drag view during column drag and drop operations.
+- `CanAutoScroll`: Returns whether auto-scrolling should happen when the column drag view reaches the left or right ends of the `SfDataGrid`.
+- `Cancel`: Returns a boolean property to cancel the event.
+
+## Cancel dragging of a particular column
+
+Dragging of a particular column can be canceled using `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if (e.From == 1 && e.DraggingAction == DataGridDragAction.DragStarted) 
+    {
+        e.Cancel = true;
+    }
+} 
+{% endhighlight %}
+
+## Cancel dropping when dragging for a particular column
+
+Dropping when dragging a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.To returns the index of the current column.
+    //e.DraggingAction returns the dragging status of the column.
+    if ((e.To > 3 && e.To < 6) &&
+    (e.DraggingAction == DataGridDragAction.DragEnded || e.DraggingAction == DataGridDragAction.Dragging))
+    {
+        e.Cancel = true;
+    }
+}
+{% endhighlight %}
+
+## Cancel dropping for a particular column
+
+The dropping of a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if (e.From == 1 && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel dropping at a particular position
+
+Dropping at a particular position can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
```

Comment:
The dropping at a particular position can be canceled using the `DataGridDragAction` argument in the `QueryColumnDragging` event.

Reply:
Modified